app/controllers/mixins/ems_common_angular.rb

```diff
@@ -171,6 +173,7 @@ def ems_form_fields
       if @ems.connection_configurations.metrics.try(:endpoint)
         metrics_hostname = @ems.connection_configurations.metrics.endpoint.hostname
         metrics_port = @ems.connection_configurations.metrics.endpoint.port
+        metrics_database_name = @ems.connection_configurations.metrics.endpoint.database_name
```

Comment:
this runs only if @ems.connection_configurations.metrics.try(:endpoint) which means its specific for the metrics endpoint which only RHEVM has anyway. So if the code runs it is for an endpoint that has database_name. (Actually all endpoint have it now, though it is empty, which is related to your second point)